### PR TITLE
Fix size of modal dialogs at different widths

### PR DESCRIPTION
**Pull request by namiltd**

### CODE REVIEW

**Comment by namiltd** [comment]
Correctly displays .modal-xl for @media min-width: between 992px and 1199px
and .modal-sm for @media min-width: less than 576px

File: scss/_modal.scss [hunk 1]
```diff
@@ -150,6 +150,8 @@
   overflow: scroll;
 }
 
+.modal-sm { max-width: $modal-sm; }
```
Comment (andresgalante) [inline scss/_modal.scss]: Why are you adding this line? it brakes the small modal on xs sizes.

**Review by andresgalante — CHANGES_REQUESTED** [review]
@namiltd thanks, I think this makes sense, other than the sm modal is broken, but just by removing line 153 works.

@mdo this PR makes the xl modal goes from xl, to lg, to sm, instead of just jumping from xl to sm. I think it makes sense.

File: scss/_modal.scss (hunk 1 above)
Comment (namiltd) [reply scss/_modal.scss]: My tests
modal-sm on a wide screen:
normal

modal-sm on a narrow screen, original:
original

modal-sm on a narrow screen, patched:
patched

Comment (andresgalante) [reply scss/_modal.scss]: Yes, I'd say that the result is accurate, we want it full screen on small viewports.

I'd remove that line.

Comment (namiltd) [reply scss/_modal.scss]: ok

Comment (namiltd) [reply scss/_modal.scss]: Ok, line removed.

**Review by namiltd — COMMENTED** [review]
Fix for .modal-sm removed.

**Review by andresgalante — APPROVED** [review]

**Comment by andresgalante** [comment]
Thanks @namiltd!

**Comment by XhmikosR** [comment]
@MartijnCuppens: LGTY?

File: scss/_modal.scss [hunk 2]
```diff
@@ -175,7 +175,8 @@
 }
 
 @include media-breakpoint-up(lg) {
-  .modal-lg { max-width: $modal-lg; }
+  .modal-lg,
+  .modal-xl { max-width: $modal-lg; }
```
Comment (MartijnCuppens) [inline scss/_modal.scss]: @XhmikosR, I would add some line breaks here:
```suggestion
  .modal-xl { 
    max-width: $modal-lg;
  }
```

Apart from that, this PR looks fine.

**Review by MartijnCuppens — CHANGES_REQUESTED** [review]

File: scss/_modal.scss (hunk 2 above)
Comment (XhmikosR) [reply scss/_modal.scss]: Agreed. I personally never use one line expressions but here on Bootstrap I didn't enforce it. We might revisit later :)

**Pull request merged**